ORStools/proc/base_processing_algorithm.py

```diff
@@ -192,11 +192,11 @@ def _get_ors_client_from_provider(provider: str, feedback: QgsProcessingFeedback
     def parseOptions(self, parameters: dict, context: QgsProcessingContext) -> dict:
         options = dict()
 
-        if parameters[self.IN_AVOID_FEATS]:
+        if parameters[self.IN_AVOID_FEATS] is not None:
```

Comment:
What is the this part for? Was there an issue without it?
As far as i tested, this only adds avoid features to the options even if it is an empty array.

Reply:
I don't remember if there was an issue without it.
It's there to avoid the problem of different values evaluating to false, while we're only interested in presence.

Reply:
I would leave it out, as we are using the same approach (without `is not None` check) in all other parameter checks.
And especially with the `[]` case where no avoid features are passed, it is still added to the request as `[]` results in `false` while `[] is not None` results in `true`. Or is there a specific case where this is definitely needed?

Reply:
no, I don't think so. Also, I don't think different values evaluating to false should be an issue.
I've cleaned this up.